eng/common/pipelines/templates/steps/verify-links.yml

```diff
@@ -1,19 +1,26 @@
 parameters:
   Directory: 'not-specified'
   IgnoreLinksFile: "$(Build.SourcesDirectory)/eng/ignore-links.txt"
-  
+  WorkingDirectory: "$(System.DefaultWorkingDirectory)"
+  ScriptDirectory: "eng/common/scripts"
+  Urls: "$(dir -r -i *.md)"
+  Recursive: $false
+  Branchreplaceregex: "$(System.PullRequest.SourceRepositoryURI)"
+  Branchreplacementname: "$(System.PullRequest.SourceCommitId)"
+  Checklinkguidance: $true
 steps:
-- task: PowerShell@2
-  displayName: Link verification check
-  inputs:
-    pwsh: true
-    workingDirectory: $(Build.SourcesDirectory)/${{ parameters.Directory }}
-    filePath: eng/common/scripts/Verify-Links.ps1 
-    arguments: >
-      -urls $(dir -r -i *.md) 
-      -rootUrl "file://$(Build.SourcesDirectory)/${{ parameters.Directory }}" 
-      -recursive: $false 
-      -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
-      -branchReplaceRegex "^($env:SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI.*/(?:blob|tree)/)master(/.*)$"
-      -branchReplacementName $env:SYSTEM_PULLREQUEST_SOURCECOMMITID
-      -devOpsLogging: $true
+  - task: PowerShell@2
+    displayName: Link verification check
+    inputs:
+      pwsh: true
+      workingDirectory: "${{ parameters.WorkingDirectory }}/${{ parameters.Directory }}"
+      filePath: ${{ parameters.ScriptDirectory }}/Verify-Links.ps1
+      arguments: >
+        -urls ${{ parameters.Urls }}
+        -rootUrl "file://${{ parameters.WorkingDirectory }}/${{ parameters.Directory }}" 
```

Comment:
With powershell, always use single quotes where you don't want there to be any change that a variable or command sneaks in, malicious or not. Same goes for the regex below: single quotes disable escaping.